### PR TITLE
Added new graph enrichment data structure

### PR DESCRIPTION
**Pull request by ganeshrvel**

### CODE REVIEW

File: playbooks/robusta_playbooks/oom_killer.py [hunk 1]
```diff
@@ -93,21 +105,36 @@ def pod_oom_killer_enricher(event: PodEvent, params: OomKillParams):
...
+        )
+        node_labels.append((
             "Node allocated memory",
             f"{allocated_precent:.2f}% out of {allocatable_memory}MB allocatable",
-        )
-        labels.append(node_label)
+        ))
+
+        if params.node_memory_graph:
+            node_graph = create_node_graph_enrichment(params, node)
```
Comment (arikalon1) [inline playbooks/robusta_playbooks/oom_killer.py:130]: Only the graph should be controlled by the `node_memory_graph` parameter
 The table should always be added

File: playbooks/robusta_playbooks/oom_killer.py [hunk 2]
```diff
@@ -116,23 +143,31 @@ def pod_oom_killer_enricher(event: PodEvent, params: OomKillParams):
...
-    table_block = TableBlock(
-        [[k, v] for (k, v) in labels],
-        ["field", "value"],
-        table_name="*Pod and Node OOMKilled data*",
-    )
-    finding.add_enrichment([table_block])
+            container_labels.append(("Container finished at", oom_killed_status.terminated.finishedAt))
+
+        if params.container_memory_graph:
+            table_block = TableBlock(
```
Comment (arikalon1) [inline playbooks/robusta_playbooks/oom_killer.py:161]: Only the graph should be controlled by the `container_memory_graph` parameter
 The table should always be added

File: src/robusta/core/model/base_params.py [hunk 3]
```diff
@@ -325,5 +327,3 @@ class LogEnricherParams(ActionParams):
     filter_regex: Optional[str] = None
 
 
-class OomKillParams(ActionParams):
```
Comment (arikalon1) [inline src/robusta/core/model/base_params.py:328]: Why did you remove it from here?

File: src/robusta/core/playbooks/prometheus_enrichment_utils.py [hunk 4]
```diff
@@ -219,11 +219,20 @@ def create_chart_from_prometheus_query(
                                            'linejoin': 'round'},)
         plot_data_list.append(plot_data)
 
+    vertical_lines = []
+    horizontal_lines = []
     for line in lines:
```
Comment (arikalon1) [inline src/robusta/core/playbooks/prometheus_enrichment_utils.py:224]: Why do we need this iteration?
No one is really using it, right?

Can we add the horizontal/vertical lines on the iteration on line 236?

File: src/robusta/core/reporting/blocks.py [hunk 5]
```diff
@@ -467,13 +473,28 @@ class PrometheusBlock(BaseBlock):
...
         """
         :param data: the PrometheusQueryResult generated created from a prometheus query
         :param query: the Prometheus query run
         """
         metadata = {"query-result-version": "1.0", "query": query}
-        super().__init__(data=data, metadata=metadata)
+        super().__init__(data=data, metadata=metadata, vertical_lines=vertical_lines, horizontal_lines=horizontal_lines,
+                         y_axis_type=y_axis_type, graph_name=graph_name)
+
+    def dict(self, *args, **kwargs) -> Dict[str, Any]:
```
Comment (arikalon1) [inline src/robusta/core/reporting/blocks.py:493]: Why do we need this?

**Review by arikalon1 — COMMENTED** [review]
nice work
left some comments

File: src/robusta/core/reporting/blocks.py (hunk 5 above)
Comment (ganeshrvel) [reply src/robusta/core/reporting/blocks.py:493]: ChartValuesFormat is not serializable. so I had to write this helper

File: src/robusta/core/playbooks/prometheus_enrichment_utils.py (hunk 4 above)
Comment (ganeshrvel) [reply src/robusta/core/playbooks/prometheus_enrichment_utils.py:224]: we already use this iteration for finding the max_y_axis which we later use in plotting the oom kill line.

File: playbooks/robusta_playbooks/oom_killer.py [hunk 6]
```diff
@@ -57,6 +59,18 @@ class OOMGraphEnricherParams(ResourceGraphEnricherParams):
     delay_graph_s: int = 0
 
 
+class OomKillParams(OOMGraphEnricherParams):
+    attach_logs: Optional[bool] = False
+    container_memory_graph: Optional[bool] = False
+    node_memory_graph: Optional[bool] = False
+
+
+def get_oomkilled_graph(oomkilled_container: PodContainer, pod: Pod, params: OOMGraphEnricherParams):
```
[the remark this answers is not in the document]
Comment (ganeshrvel) [reply playbooks/robusta_playbooks/oom_killer.py]: done

File: playbooks/robusta_playbooks/oom_killer.py [hunk 7]
```diff
@@ -93,21 +105,36 @@ def pod_oom_killer_enricher(event: PodEvent, params: OomKillParams):
...
-    node: Node = Node.readNode(pod.spec.nodeName).obj  # type: ignore
+
     if node:
         allocatable_memory = PodResources.parse_mem(node.status.allocatable.get("memory", "0Mi"))
         capacity_memory = PodResources.parse_mem(node.status.capacity.get("memory", "0Mi"))
         allocated_precent = (capacity_memory - allocatable_memory) * 100 / capacity_memory
-        node_label = (
+
+        node_labels: List[Tuple[str, str]] = []
+        node_labels.extend(labels)
```
[the remark this answers is not in the document]
Comment (ganeshrvel) [reply playbooks/robusta_playbooks/oom_killer.py]: done

File: playbooks/robusta_playbooks/oom_killer.py (hunk 1 above)
Comment (ganeshrvel) [reply playbooks/robusta_playbooks/oom_killer.py:120]: done

File: playbooks/robusta_playbooks/oom_killer.py (hunk 2 above)
Comment (ganeshrvel) [reply playbooks/robusta_playbooks/oom_killer.py]: done

File: src/robusta/core/model/base_params.py (hunk 3 above)
Comment (ganeshrvel) [reply src/robusta/core/model/base_params.py:328]: reverted

File: src/robusta/core/reporting/blocks.py [hunk 8]
```diff
@@ -467,13 +473,28 @@ class PrometheusBlock(BaseBlock):
     data: PrometheusQueryResult
     metadata: Dict[str, str]
 
-    def __init__(self, data: PrometheusQueryResult, query: str):
+    vertical_lines: Optional[List[PrometheusBlockLineData]]
+    horizontal_lines: Optional[List[PrometheusBlockLineData]]
+    y_axis_type: Optional[ChartValuesFormat]
+    graph_name: Optional[str]
+
+    def __init__(self, data: PrometheusQueryResult, query: str, vertical_lines: Optional[List[PrometheusBlockLineData]],
```
Comment (arikalon1) [inline src/robusta/core/reporting/blocks.py:481]: This should have default None to all optional fields, because this way it breaks all other usages of PrometheusBlock
```
def __init__(self, data: PrometheusQueryResult, query: str, 
vertical_lines: Optional[List[PrometheusBlockLineData]] = None,
                 horizontal_lines: Optional[List[PrometheusBlockLineData]] = None,
                 y_axis_type: Optional[ChartValuesFormat] = None,
                 graph_name: Optional[str] = None):
```

**Review by arikalon1 — APPROVED** [review]
Left one more comment
Please fix, test and merge
